### ISSUE TITLE
[zattoo] Add support for zattoo.com and its partner sites

### ISSUE THREAD

**Issue by goggle** [rg3/youtube-dl#14668]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.10.29*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2017.10.29**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):

The videos are usually only available for like 7 days! So the links probably won't work, but the scheme should be clear.
- Single video: https://zattoo.com/watch/3plus/130760870-bauer-ledig-sucht
- Live stream: https://zatto.com/watch/srf1

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

zattoo.com is a big TV streaming service provider. With a free account you can only watch the live program, with a paid account you get access to the replay functionality. All the video should be available for at least 7 days. You can make a 30 days premium account for free on zattoo.com. It's possible that some content is geo restricted.

Zattoo has several partners, which use exactly the same interface:
 - https://mobiltv.quickline.com
 - https://www.netplus.tv
 - https://tvplus.m-net.de

Would it be possible to get support for zattoo.com and its partners into youtube-dl? 

I have written a information extractor for that. As soon as it's finished, I will open a pull request.

**Comment by troych** [comment]
This seems broken atm, worked for me when I checked out the zattoo branch a couple of weeks ago, now I only get an error.

```
youtube-dl -v "https://zattoo.com/watch/srf_zwei/131484236-feuer-und-flamme/99350971/1511253900000/1511256300000/-300000" -o feuer_20.11.ts
[debug] System config: []
[debug] User config: [u'--username', u'PRIVATE', u'--password', u'PRIVATE']
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://zattoo.com/watch/srf_zwei/131484236-feuer-und-flamme/99350971/1511253900000/1511256300000/-300000', u'-o', u'feuer_20.11.ts']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.10.29
[debug] Python version 2.7.9 - Linux-3.16.0-4-amd64-x86_64-with-debian-8.10
[debug] exe versions: ffmpeg 3.2.5-1, ffprobe 3.2.5-1
[debug] Proxy map: {}
[Zattoo] Downloading https://zattoo.com
[Zattoo] Say hello
[Zattoo] Logging in
[Zattoo] 131484236: Downloading video information
[Zattoo] 131484236: Downloading dash formats
ERROR: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl-2017.10.29-py2.7.egg/youtube_dl/extractor/common.py", line 503, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl-2017.10.29-py2.7.egg/youtube_dl/YoutubeDL.py", line 2195, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```
Or could that be something on my end? I see it's timeouting somewhere, but I can watch zattoo and recordings via web.

**Comment by goggle** [comment]
@troych Unfortunately I don't have a zattoo premium account, so I can not test this issue on my site. I have an account for a zattoo partner though, but there I have only 7 days replay available, and your video seems to older than that. Have you encountered the same issue for a recent video?

**Comment by troych** [comment]
Ye the video in question is a recording, not via replay. Just tested a file within the 7 day replay period and there it works indeed. So I guess the recording bit is broken. I guess I could give you access to my zattoo account if that would help? Not sure how that works with geo restrictions and the likes though, since I'm situated in Switzerland.

**Comment by goggle** [comment]
@troych I was not aware that there is a recording feature for zattoo, because I use the zattoo partner Quickline and they don't offer this. I have just seen that I can record up to 20 shows with my free zattoo account. I am recording something now and I will test this recording with this information extractor afterwards.

**Comment by goggle** [comment]
@troych I have added support for zattoo, you just need to pull my latest commit https://github.com/rg3/youtube-dl/pull/14676/commits/ce8053b6464a3c02d1fd1c7a9d32869bb99ecb0a. Please test it with your premium account and let me know if everything works as expected.

**Comment by troych** [comment]
Just had a quick go, recording downloading as we speak. Thank you very much for the help and fix. :)

**Comment by MarS919** [comment]
I have a premium account. Downloading my Zattoo recordings works using goggle's commit ce8053b. Thanks!

**Comment by goggle** [comment]
@troych @MarS919 To get my pull request for Zattoo support into the official master branch of youtube-dl, the maintainer of youtube-dl @dstftw needs a zattoo premium account to test my code. If any of you could provide their login credentials to @dstftw for testing purposes, that would be great (see the discussion in #14676). Also @dstftw needs to have access to a Swiss IP or an IP from the country where your zattoo account is registered at (because zattoo is heavily geo-restricted), but I guess this shouldn't be a problem.
As far as I see, the one thing that should be added to my code is a correct way to handle the geo-restriction errors.

**Comment by troych** [comment]
I've contacted dstftw, thanks for the heads up.

**Comment by goggle** [comment]
@troych @dstftw Do we have any news about this?

**Comment by dstftw** [comment]
Haven't received any credentials.

**Comment by troych** [comment]
@dstftw I sent you an email on the 4th, never got an answer though?

**Comment by troych** [comment]
@goggle Can you get in contact with @dstftw, I'll be happy to lend my account for testing but I can't seem to get a response via email.

**Comment by goggle** [comment]
@troych Sure, I can try. Can you PM me your email address, so I can give it to dstftw, that he knows who you are?

**Comment by dstftw** [comment]
I guess I made it quite clear: if you want to provide account credentials send **account credentials** not requests to contact somebody at somewhere.

**Comment by troych** [comment]
Well, I sent you an email asking you about the need for vpn since zattoo is geo restricted. I was just looking for a quick sign of life so I can change my password to something I can send out. I did not want to make this overly complicated, and wasn't intending to make your access any harder then it needs to be.

I hope you understand that I didn't just want to send account credentials without knowing if a) they're actually useful to you because of geo restrictions and b) the email I wrote to is actually in use.

Sorry for any misunderstandings.
